### ISSUE TITLE
`getComputedStyle` doesn't returns correct style after I dynamically add `<style>` tag to the document whose content is set by `.innerHTML`.

### ISSUE THREAD

**Issue by StephenChips** [jsdom/jsdom#2986]
### Basic info:

- **Node.js version:** v12.13.0
- **jsdom version:** v16.2.2

I did some tests with Jest today, which use JSDOM to simulate the browser environment. I was going to dynamically inject some CSS codes after my component is loaded, so I created a `<style>` tag and set the style string to its `innerText` property. After the tag was inserted, surprisingly the test failed, and said that the style didn't get updated properly (I used the `window.getComputedStyle` function to get the current style). After some troubleshooting I didn't found any bugs, then I changed assignning style from `.innerText' to `.textContent`. The test passed  after such modification. So I though maybe it's the JSDOM's problem. I created following snippet and it proved that it was.

### Minimal reproduction case

```js
var jsdom = require('jsdom');
var jsdom = new jsdom.JSDOM('<html><head></head><body><p class="hello">hello, world.</p></body></html>');
var document = jsdom.window.document;
var window = jsdom.window;
var display = window.getComputedStyle(document.querySelector('.hello')).display;
console.log(display); // block

var style = document.createElement('style');
style.innerText = '.hello { display: flex; }';
console.log(style.innerText); // `.hello { display: flex; }
undefined
document.head.appendChild(style);

console.log(document.documentElement.outerHTML);
// <html><head><style></style></head><body><p class="hello">hello, world.</p></body></html>

display = **window.getComputedStyle(document.querySelector('.hello')).display;**
console.log(display);
// block (which should be 'flex')
```

### How does similar code behave in browsers?
The browser shows the value 'flex' as it should be.

**Comment by milahu** [comment]
yes, <tt>[window.getComputedStyle](https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle)(e [, p])</tt> should return a 'live' object

> The returned style is a **live** CSSStyleDeclaration object, which updates automatically when the element's styles are changed.

in
https://github.com/jsdom/jsdom/blob/5279cfda5fe4d52f04b2eb6a801c98d81f9b55da/lib/jsdom/browser/Window.js#L643
doing `return elt.style` is too simple
cos `elt.style` has less data than `getComputedStyle`

`elt.style` only contains data that was explicitly set
where GCS contains all data needed for layout rendering

jsdom already adds default values like `div{display: block}`
just the result is not live ....

**Comment by milahu** [comment]
> `style.innerText = '.hello { display: flex; }';`

currently we must use `style.innerHTML`, see issue #3052

**Comment by strangedev** [comment]
Is this intended behavior or a bug? If it is a bug, can we do anything to get this addressed? I'd be glad to help out.

**Comment by DanKaplanSES** [comment]
This may be a duplicate of #2160

**Comment by ianzone** [comment]
issue still exists in v26
